docs/best-practices.md

```diff
@@ -0,0 +1,14 @@
+# Best Practices
+
+Guardian CDK codifies best practices as defined in the [recommendations](https://github.com/guardian/recommendations)
+repo.
+
+Here are some best practices for using this library:
+- Use [snapshot testing with Jest](https://jestjs.io/docs/snapshot-testing). Snapshot testing makes it easy to review
+  any stack changes when developing locally and also in PRs.
```

Comment:
Should we expand this, describing why this is preferred compared to the alternative(s)? For example, snapshots allow us to catch things early, therefore it creates a shorter feedback loop.

Reply:
Worth linking to AWS's docs too? https://docs.aws.amazon.com/cdk/v2/guide/testing.html#testing_snapshot